src/freenet/clients/http/WelcomeToadlet.java

```diff
@@ -265,6 +265,18 @@ public void handleMethodPOST(URI uri, HTTPRequest request, ToadletContext ctx) t
             writeHTMLReply(ctx, 200, "OK", pageNode.generate());
             request.freeParts();
             bucket.free();
+            return;
+        } else if (request.isPartSet("key")) {
+            if(!ctx.checkFormPassword(request)) return;
+            String key;
+            try {
+              key = URLDecoder.decode(new FreenetURI(request.getPartAsStringFailsafe("key", 256)).toURI("/").toString(), false);
+            } catch (Exception e) {
+              sendErrorPage(ctx, "Invalid URI", "This URI is invalid!", e);
```

Comment:
I know that we've translated error messages in the past... but I'm not sure it's a good idea; do you insist on this?